napari/plugins/tests/fixtures/napari_bad_plugin.py

```diff
@@ -0,0 +1,18 @@
+"""
+Test plugin that is technically correct, but would cause performance problems
+"""
+import pluggy
+import time
+
+hookimpl = pluggy.HookimplMarker("napari")
+
+
+def reader_function():
+    raise IOError('it worked')
+
+
+@hookimpl
+def napari_get_reader(path):
```

Comment:
This plugin isn't tested yet right? (Which is fine, just want to make sure I'm not missing it)

Reply:
correct.  can delay that for the next PR (where tests will also be expanded).    #935 also sort of tests this (but not via pytest).

Reply:
I don't mind, as I said I'm fine with how this is all trending and this is easy to change in the future too, so I see no harm in it

Reply:
It's impossible to test without at least one implementation... and that is introduced in #935.  So #935 should not be approved/merged until everyone is happy with the level of testing there.  Other than one hookspec for the purpose of testing, I'm trying to keep all of the reader-plugin specific stuff out of this PR.  There's plenty to do in #935 though

Reply:
:+1: